### PR TITLE
Manual merge of a TMT and GFW records is not working 

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/VV-341

### CODE REVIEW

File: apps/vessel-history/features/profile/Profile.tsx [hunk 1]
```diff
@@ -62,15 +64,29 @@ const Profile: React.FC = (props): React.ReactElement => {
...
+      if (dataset.toLocaleLowerCase() === 'na') {
+        const gfwAka = akaVesselProfileIds.find(aka => {
+          const [akaDataset] = aka.split('_')
+          return akaDataset.toLocaleLowerCase() !== 'na'
+        })
+        if (gfwAka) {
+          const [akaDataset, akaGfwId] = gfwAka.split('_')
+          dataset = akaDataset
+          gfwId = akaGfwId
+        }
```
Comment (rdgfuentes) [inline apps/vessel-history/features/profile/Profile.tsx:80]: 👍

File: apps/vessel-history/features/profile/Profile.tsx [hunk 2]
```diff
@@ -87,7 +103,9 @@ const Profile: React.FC = (props): React.ReactElement => {
             // Only merge with vessels of the same dataset that the main vessel
             const akaVesselsIds = (akaVesselProfileIds ?? [])
               .map((vesselProfileId) => parseVesselProfileId(vesselProfileId))
-              .filter((akaVessel) => akaVessel.dataset === dataset && akaVessel.id)
+            //.filter((akaVessel) => akaVessel.dataset === dataset && akaVessel.id)
```
Comment (rdgfuentes) [inline apps/vessel-history/features/profile/Profile.tsx:106]: Remove it if it's not necessary

File: apps/vessel-history/features/profile/Profile.tsx [hunk 3]
```diff
@@ -121,6 +139,8 @@ const Profile: React.FC = (props): React.ReactElement => {
   }, [dispatchLocation, query])
 
   useEffect(() => {
+    console.log(resourceQueries)
+    console.log(vesselDataviewLoaded)
```
Comment (rdgfuentes) [inline apps/vessel-history/features/profile/Profile.tsx:143]: 🧹

File: apps/vessel-history/features/vessels/sources/gfw.slice.ts [hunk 4]
```diff
@@ -7,8 +7,22 @@ interface GFWVesselSourceId extends VesselSourceId {
...
+  byCount: byCount,
+  byDate: data[field] ? [
+    {
+      value: data[field],
+      endDate: data.lastTransmissionDate,
+      firstSeen: data.firstTransmissionDate,
+      source: VesselAPISource.GFW,
+    },
+  ] : []
+})
```
Comment (rdgfuentes) [inline apps/vessel-history/features/vessels/sources/gfw.slice.ts:22]: Nice

File: apps/vessel-history/features/vessels/sources/gfw.slice.ts [hunk 5]
```diff
@@ -23,40 +37,18 @@ export const toVessel: (data: GFWDetail) => VesselWithHistory = (data: GFWDetail
...
-            endDate: data.lastTransmissionDate,
-            firstSeen: data.firstTransmissionDate,
-            source: VesselAPISource.GFW,
-          },
-        ],
-      },
-      operator: emptyHistory,
+      callsign: getHistoryField(data, 'callsign', data.otherCallsigns),
+      geartype: getHistoryField(data, 'geartype'),
+      imo: getHistoryField(data, 'imo'),
```
Comment (rdgfuentes) [inline apps/vessel-history/features/vessels/sources/gfw.slice.ts:42]: `data.otherImos` should be assigned to the `byCount` arg, right?

File: apps/vessel-history/features/vessels/sources/tmt.slice.ts [hunk 6]
```diff
@@ -15,6 +15,10 @@ const extractValue: (valueItem: ValueItem[]) => string | undefined = (valueItem:
 const sortAuthorization = (a: Authorization, b: Authorization) =>
   a.originalStartDate > b.originalStartDate ? 1 : -1
 
+const getHistoryField = (historyField: AnyValueList[]): VesselFieldHistory<any> => ({
```
Comment (rdgfuentes) [inline apps/vessel-history/features/vessels/sources/tmt.slice.ts:18]: 👍

File: apps/vessel-history/features/vessels/vessels.utils.ts [hunk 7]
```diff
@@ -109,14 +84,7 @@ const mergeHistoryFields = (
...
             byCount: (acc[fieldName]?.byCount || []).concat(current[fieldName].byCount),
-            byDate: (acc[fieldName]?.byDate || []).concat(
-              getFieldHistory<ValueItem>(
-                fieldName,
-                current[fieldName]?.byDate ?? [],
-                dataValues[sourceIndex].vessel,
-                dataValues[sourceIndex].source
-              )
-            ),
+            byDate: (acc[fieldName]?.byDate || []).concat(current[fieldName].byDate),
```
Comment (rdgfuentes) [inline apps/vessel-history/features/vessels/vessels.utils.ts:87]: Much cleaner now 👏

File: apps/vessel-history/features/vessels/sources/gfw.slice.ts (hunk 5 above)
Comment (fertandil87) [reply apps/vessel-history/features/vessels/sources/gfw.slice.ts:42]: yes

**Pull request merged**